torch/testing/_internal/common_utils.py

```diff
@@ -939,6 +938,8 @@ def compare_with_numpy(self, torch_fn, np_fn, tensor_like,
                 # NOTE: copying an array before conversion is necessary when,
                 #   for example, the array has negative strides.
                 np_result = torch.from_numpy(np_result.copy())
+            if dtype is torch.bfloat16 and np_result.dtype is torch.float:
+                np_result = np_result.bfloat16()
```

Comment:
should this convert pytorch result to `float` instead?

Reply:
Is this change actually necessary as part of this PR?

We generally don't want to convert the NumPy result back to bfloat16 because that introduces an additional source of error.

Reply:
I have changed this to convert the torch result to fp32. A cast here is necessary, otherwise, this test will fail with `exact_dtype` error